install/uninstall.ps1

```diff
@@ -0,0 +1,125 @@
+# Enable a default -Verbose flag for debug output.
+[CmdletBinding()]
+
+# Halt script execution at the first failed command.
+$script:ErrorActionPreference='Stop'
+
+$azureauthDefaultLocation = ([System.IO.Path]::Combine($Env:LOCALAPPDATA, "Programs", "AzureAuth"))
+
+if (![string]::IsNullOrEmpty($Env:AZUREAUTH_INSTALL_DIRECTORY) `
+        -And $Env:AZUREAUTH_INSTALL_DIRECTORY -ne $azureauthDefaultLocation) {
+    Write-Warning "Ignoring AZUREAUTH_INSTALL_DIRECTORY environment variable. Custom uninstallation locations are not supported."
+}
+
+function Uninstall {
+    Close-AzureauthInstances
+
+    $locations = Get-AzureauthsInPath
+    Remove-AzureauthDirectories $locations
+    Remove-FromPath $locations
+
+    Write-Output "Uninstalled AzureAuth!"
+}
+
+function Get-AzureauthsInPath {
+    $allLocations = [System.Collections.ArrayList]@()
+    $azureauthsInPath = (Get-Command -Name azureauth -ErrorAction SilentlyContinue -CommandType Application -All).Source
+
+    # Uninstallation is only supported from the default AzureAuth location.
+    # We warn the user of any custom locations that are listed in the PATH.
+    # Installations in custom locations that are not listed in PATH cannot
+    # be found and uninstalled.
+    # We still...
+    ForEach($az in $azureauthsInPath) {
+        $tempDirectory = (Get-Item $az).Directory.FullName
+        if (!$az.Contains($azureauthDefaultLocation)) {
+            Write-Warning "Additional installation found in ${tempDirectory}"

```

Comment:
Should we expect users to take any action in this case? If so, we should print that. Additionally, I see you have added a comment `Installations in custom locations that are not listed in PATH cannot be found and uninstalled.` I think we should explicitly warn the user with this comment if there are no actions they need to take.

Reply:
I added the message at the end that @reillysiemens suggested.